spark_auto_mapper_fhir/resources/immunization.py

```diff
@@ -1,30 +1,172 @@
+from spark_auto_mapper_fhir.complex_types.simple_quantity import SimpleQuantity
+from spark_auto_mapper_fhir.fhir_types.boolean import FhirBoolean
+from spark_auto_mapper_fhir.fhir_types.date import FhirDate
+from spark_auto_mapper_fhir.fhir_types.date_time import FhirDateTime
 from spark_auto_mapper_fhir.fhir_types.id import FhirId
 from spark_auto_mapper_fhir.fhir_types.list import FhirList
+from spark_auto_mapper_fhir.fhir_types.string import FhirString
+from spark_auto_mapper_fhir.resources.condition import Condition
+from spark_auto_mapper_fhir.resources.diagnostic_report import DiagnosticReport
+from spark_auto_mapper_fhir.resources.encounter import Encounter
 from spark_auto_mapper_fhir.resources.fhir_resource_base import FhirResourceBase
 from spark_auto_mapper_fhir.extensions.extension import Extension
 from spark_auto_mapper_fhir.complex_types.meta import Meta
+from spark_auto_mapper_fhir.resources.location import Location
+from spark_auto_mapper_fhir.resources.observation import Observation
+from spark_auto_mapper_fhir.resources.organization import Organization
+from spark_auto_mapper_fhir.resources.patient import Patient
+from spark_auto_mapper_fhir.valuesets.immunization_function import (
+    ImmunizationFunctionCode,
+)
+from spark_auto_mapper_fhir.valuesets.immunization_funding_source import (
+    ImmunizationFundingSourceCode,
+)
+from spark_auto_mapper_fhir.valuesets.immunization_origin import ImmunizationOriginCode
+from spark_auto_mapper_fhir.valuesets.immunization_program_elegibility import (
+    ImmunizationProgramElegibilityCode,
+)
+from spark_auto_mapper_fhir.valuesets.immunization_reason_code import (
+    ImmunizationReasonCode,
+)
+from spark_auto_mapper_fhir.valuesets.immunization_route import ImmunizationRouteCode
+from spark_auto_mapper_fhir.valuesets.immunization_site import ImmunizationSiteCode
+
+from spark_auto_mapper_fhir.valuesets.immunization_status_reason import (
+    ImmunizationStatusReasonCode,
+)
+from spark_auto_mapper_fhir.complex_types.identifier import Identifier
+from spark_auto_mapper_fhir.valuesets.immunization_subpotent_reason import (
+    ImmunizationSubpotentReasonCode,
+)
+from spark_auto_mapper_fhir.valuesets.immunization_target_disease import (
+    ImmunizationTargetDiseaseCode,
+)
+from spark_auto_mapper_fhir.valuesets.vaccine_code import VaccineCode
 
 
 class Immunization(FhirResourceBase):
     def __init__(
         self,
+        *,
         id_: Optional[FhirId] = None,
         meta: Optional[Meta] = None,
         extension: Optional[FhirList[Extension]] = None,
+        identifier: Optional[FhirList[Identifier]] = None,
+        status: ImmunizationSiteCode,
+        statusReason: Optional[CodeableConcept[ImmunizationStatusReasonCode]] = None,
+        vaccineCode: CodeableConcept[VaccineCode],
+        patient: Reference[Patient],
+        encounter: Optional[Reference[Encounter]] = None,
+        occurrenceDateTime: FhirDateTime,
+        occurrenceString: FhirString,
```

Comment:
same as doseNumber, both can't be mandatory

Reply:
Done